tests/E2E Tests/WebAppUiTests/B2CWebAppCallsWebApiLocally.cs

```diff
@@ -77,16 +70,41 @@ public async Task Susi_B2C_LocalAccount_TodoAppFucntionsCorrectly()
             IBrowserContext context = await browser.NewContextAsync(new BrowserNewContextOptions { IgnoreHTTPSErrors = true });
             await context.Tracing.StartAsync(new() { Screenshots = true, Snapshots = true, Sources = true });
 
+            Process? serviceProcess= null;
+            Process? clientProcess = null;
+
             try
             {
+                // Start the web app and api processes.
+                // The delay before starting client prevents transient devbox issue where the client fails to load the first time after rebuilding.
+                // The delay after processes are started gives time to finish initial setup before attempted connection.
+                serviceProcess = UiTestHelpers.StartProcessLocally(_testAssemblyPath, _devAppPath + TC.s_todoListServicePath, TC.s_todoListServiceExe, serviceEnvVars);
+                Thread.Sleep(3000);
+                clientProcess = UiTestHelpers.StartProcessLocally(_testAssemblyPath, _devAppPath + TC.s_todoListClientPath, TC.s_todoListClientExe, clientEnvVars);
+                Thread.Sleep(5000);
+
                 if (!UiTestHelpers.ProcessesAreAlive(new List<Process>() { clientProcess, serviceProcess }))
                 {
                     Assert.Fail(TC.WebAppCrashedString);
                 }
 
-                // Navigate to web app
+                // Navigate to web app the retry logic ensures the web app has time to start up to establish a connection.
                 IPage page = await context.NewPageAsync();
-                await page.GotoAsync(TC.LocalhostUrl + TodoListClientPort);
+                uint InitialConnectionRetryCount = 5;
+                while (InitialConnectionRetryCount > 0)
+                {
+                    try
+                    {
+                        await page.GotoAsync(TC.LocalhostUrl + TodoListClientPort);
+                        break;
+                    }
+                    catch (PlaywrightException ex)
+                    {
+                        Thread.Sleep(1000);
```

Comment:
Don't use Thraed.Sleep for waiting. Use `await Task.Delay(1000)`. Thread.Sleep results in spin waiting. Task.Delay will cause .NET to pause your thread and give the CPU time to another thread. Might be quite helpful if you run tests in parallel.

Reply:
changed